text/0000-sizeof-alignof-offsetof-typeof.md

```diff
@@ -0,0 +1,87 @@
+- Start Date: 2015-01-17
+- RFC PR #: (leave this empty)
+- Rust Issue #: (leave this empty)
+
+# Summary
+
+Add `sizeof`, `alignof` and `offsetof` operators similar to the ones found in C/C++.  
+Add `typeof` operator for finding out the type of an expression or a struct field.
+
+# Motivation
+
+- Currently it is not possible to use size / alignment of a type in const expressions.
+- It is not possible to find out size / alignment of a struct field without creating an instance of that struct.
+- The equivalent of C's `offsetof(struct, field)` is not supported at all.
+
+These are very annoying shortcomings for a "systems" language, so let's fix them!
+
+# Detailed design
+
+This RFC proposes to add the following operators:
+
+| Operator            | Evaluates to...  |
+|---------------------|------------------|
+|`sizeof(T)`          |The size of type T.|
+|`alignof(T)`         |The minimal alignment of type T.|
+|`offsetof(F in S)`   |Offset of field F in struct S. F may be either an identifier for regular structs or an integer index for tuples and tuple structs.|
+|`typeof(F in S)`     |The type of field F in struct S. F may be either an identifier for regular structs or an integer index for tuples and tuple structs.|
```

Comment:
I suspect this may stop `x in y` ever becoming an expression (I don't know if we want to do this at all, but...).